File: backend/internal/model/cron_task.go
package model

import (
	"database/sql"
	"time"

	"gorm.io/gorm"
)

// TaskStatus 定时任务状态
type TaskStatus string

const (
	TaskStatusPending   TaskStatus = "pending"   // 待执行
	TaskStatusRunning   TaskStatus = "running"   // 执行中
	TaskStatusCompleted TaskStatus = "completed" // 已完成（once 类型）
	TaskStatusStopped   TaskStatus = "stopped"   // 已停止
	TaskStatusFailed    TaskStatus = "failed"    // 失败
)

// TaskType 定时任务类型
type TaskType string

const (
	TaskTypeOnce      TaskType = "once"      // 一次性任务
	TaskTypeRecurring TaskType = "recurring" // 重复执行任务
)

// CronTask 定时任务
type CronTask struct {
	ID uint `gorm:"primarykey"`

	Name            string     `gorm:"type:varchar(100);not null;comment:'任务名称'"`
	Spec            string     `gorm:"type:varchar(50);comment:'cron 表达式'"`
	Type            TaskType   `gorm:"type:varchar(20);not null;comment:'任务类型 (once/recurring)'"`
	Status          TaskStatus `gorm:"type:varchar(20);not null;default:pending;comment:'任务状态'"`
	ExecType        string     `gorm:"type:varchar(50);not null;comment:'执行类型'"`
	Raw             string     `gorm:"type:text;comment:'原始数据 (JSON 格式)'"`
	ValidFrom       sql.NullTime `gorm:"comment:'有效期开始'"`
	ValidUntil      sql.NullTime `gorm:"comment:'有效期结束'"`
	Enabled         bool       `gorm:"not null;default:true;comment:'是否启用'"`
	MaxRetries      int        `gorm:"not null;default:0;comment:'最大重试次数'"`
	LastExecutedAt  sql.NullTime `gorm:"comment:'上次执行时间'"`
	NextExecutionAt sql.NullTime `gorm:"comment:'下次执行时间'"`
	TotalExecutions uint       `gorm:"not null;default:0;comment:'累计执行次数'"`

	CreatedAt time.Time `gorm:"index:,sort:desc;"`
	UpdatedAt time.Time
	DeletedAt gorm.DeletedAt `gorm:"index"`
}

func (CronTask) TableName() string {
	return "cron_task"
}

File: frontend/src/types/crontask.ts
/**
 * 定时任务相关类型定义
 */

/**
 * 任务状态
 */
export type TaskStatus = 'pending' | 'running' | 'completed' | 'stopped' | 'failed';

/**
 * 任务类型
 */
export type TaskType = 'once' | 'recurring';

/**
 * 执行类型
 */
export type ExecType = 'OKXWatcher';

/**
 * 定时任务详情
 */
export interface CronTask {
  id: number;
  name: string;
  spec: string;
  type: TaskType;
  status: TaskStatus;
  exec_type: ExecType;
  raw: string; // JSON 字符串：{"symbol": "ETH-USDT-SWAP"}
  valid_from: string;
  valid_until: string;
  enabled: boolean;
  max_retries: number;
  timeout_seconds: number;
  last_executed_at?: string;
  next_execution_at?: string;
  total_executions: number;
  created_at: string;
  updated_at: string;
}

/**
 * 创建任务请求
 */
export interface CreateTaskRequest {
  name: string;
  spec: string;
  type: TaskType;
  exec_type: ExecType;
  raw: string;
  valid_from: string;
  valid_until: string;
  enabled: boolean;
  max_retries?: number;
  timeout_seconds?: number;
  next_execution_at?: string;
}

/**
 * 更新任务请求
 */
export interface UpdateTaskRequest {
  name?: string;
  spec?: string;
  type?: TaskType;
  exec_type?: ExecType;
  raw?: string;
  valid_from?: string;
  valid_until?: string;
  enabled?: boolean;
  max_retries?: number;
  timeout_seconds?: number;
  next_execution_at?: string;
}

/**
 * 启动任务请求
 */
export interface StartTaskRequest {
  next_execution_time: string;
}

/**
 * 任务列表查询参数
 */
export interface TaskListParams {
  page?: number;
  pageSize?: number;
  status?: TaskStatus;
  type?: TaskType;
  enabled?: boolean;
}
